pkg/user/password_reset.go

```diff
@@ -0,0 +1,42 @@
+package user
+
+import (
+	"time"
+
+	"github.com/dgryski/trifles/uuid"
+)
+
+type PasswordReset struct {
+	Token     string
+	CreatedAt time.Time
+}
+
+func NewPasswordReset() *PasswordReset {
+	return &PasswordReset{
+		Token:     generateToken(),
+		CreatedAt: time.Now(),
+	}
+}
+
+func PasswordResetFrom(token string, createdAt time.Time) *PasswordReset {
+	return &PasswordReset{
+		Token:     token,
+		CreatedAt: createdAt,
+	}
+}
+
+func generateToken() string {
+	return uuid.UUIDv4()
+}
+
+func (pr *PasswordReset) Valid(token string) bool {
```

Comment:
Sorry, I made a mistake and thought this method was just a simple getter, but it accepts an arg and has logic. I think `Validate` is better name

```suggestion
func (pr *PasswordReset) Validate(token string) bool {
```